pkg/pipeline/ConfigMapService.go

```diff
@@ -1784,3 +1802,139 @@ func (impl ConfigMapServiceImpl) buildBulkPayload(bulkPatchRequest *BulkPatchReq
+			if err != nil {
+				impl.logger.Errorw("error while creating env level", "error", err)
+				return nil, err
+			}
+			return createJobEnvOverrideRequest, nil
+		}
+		env, err := impl.environmentRepository.FindById(configMap.EnvironmentId)
+		if err != nil {
+			impl.logger.Errorw("error while fetching environment from db", "error", err)
+			return nil, err
+		}
+		impl.logger.Warnw("Environment override in this environment already exits", "appId", createJobEnvOverrideRequest.AppId, "envId", createJobEnvOverrideRequest.EnvId)
+		return nil, errors.New("Environment " + env.Name + " already exists.")
+	}
+	model := &chartConfig.ConfigMapEnvModel{
+		AppId:         createJobEnvOverrideRequest.AppId,
+		EnvironmentId: createJobEnvOverrideRequest.EnvId,
+		Deleted:       false,
+	}
+	model.CreatedBy = createJobEnvOverrideRequest.UserId
+	model.UpdatedBy = createJobEnvOverrideRequest.UserId
+	configMap, err = impl.configMapRepository.CreateEnvLevel(model)
+	if err != nil {
+		impl.logger.Errorw("error while creating env level", "error", err)
+		return nil, err
+	}
+	return createJobEnvOverrideRequest, nil
+}
+
+func (impl ConfigMapServiceImpl) ConfigSecretEnvironmentDelete(createJobEnvOverrideRequest *CreateJobEnvOverridePayload) (*CreateJobEnvOverridePayload, error) {
+	configMap, err := impl.configMapRepository.GetByAppIdAndEnvIdEnvLevel(createJobEnvOverrideRequest.AppId, createJobEnvOverrideRequest.EnvId)
+	if pg.ErrNoRows == err {
+		impl.logger.Warnw("Environment override in this environment doesn't exits", "appId", createJobEnvOverrideRequest.AppId, "envId", createJobEnvOverrideRequest.EnvId)
+		return nil, err
+	}
+	if err != nil {
+		impl.logger.Errorw("error while fetching from db", "error", err)
+		return nil, err
+	}
+	configMap.Deleted = true
+	configMap.ConfigMapData = ""
+	configMap.SecretData = ""
+	configMap.UpdatedBy = createJobEnvOverrideRequest.UserId
+	_, err = impl.configMapRepository.UpdateEnvLevel(configMap)
+	if err != nil {
+		impl.logger.Errorw("error while updating env level", "error", err)
+		return nil, err
+	}
+	return createJobEnvOverrideRequest, nil
+}
+
+func (impl ConfigMapServiceImpl) ConfigSecretEnvironmentGet(appId int) ([]JobEnvOverrideResponse, error) {
+	configMap, err := impl.configMapRepository.GetEnvLevelByAppId(appId)
+	if err != nil {
+		impl.logger.Errorw("error while fetching envConfig from db", "error", err)
+		return nil, err
+	}
+	var envIds []*int
+	for _, cm := range configMap {
+		envIds = append(envIds, &cm.EnvironmentId)
+	}
+	var jobEnvOverrideResponse []JobEnvOverrideResponse
+
+	if len(envIds) == 0 {
+		return jobEnvOverrideResponse, nil
+	}
+	envs, err := impl.environmentRepository.FindByIds(envIds)
+
+	if err != nil {
+		impl.logger.Errorw("error while fetching environments from db", "error", err)
+		return nil, err
+	}
+
+	envIdNameMap := make(map[int]string)
+
+	for _, env := range envs {
+
+		envIdNameMap[env.Id] = env.Name
+	}
```

Comment:
this is similar to my other comments. Please refrain from rewriting the same logic at different places